shelley/chain-and-ledger/formal-spec/epoch.tex

```diff
@@ -810,6 +825,31 @@ \subsection{Complete Epoch Boundary Transition}
 
 The epoch transition rule calls $\mathsf{SNAP}$, $\mathsf{POOLREAP}$ and $\mathsf{NEWPP}$
 in sequence.
+Figure~\ref{fig:funcs:epoch-helper} gives some helper functions for the
+$\mathsf{EPOCH}$ transition.
+
+%%
+%% Figure - Helper Functions for Epoch Rules
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{pvotes} \in \Update \to \Ppm \to (\VKeyGen \mapsto \Value) \\
+      & \fun{pvotes}~\var{up}~\var{ppm}
+      = \{k\mapsto v~\mid~ k\mapsto\var{ppm} ~\land~ \var{ppm}\mapsto v\}\\
+  \end{align*}
+
+  \begin{align*}
+      & \fun{tally} \in \Update \to \PParams\\
+      & \fun{tally}~\var{up} =
+          \Big\{\var{ppm}\mapsto v
+          ~\Big|~
+          |\fun{pvotes}~\var{up}~\var{ppm}\restrictrange\var{v}|\geq 5 \Big\}
+  \end{align*}
```

Comment:
I'm also wondering, do we need the flexibility of updating individual values? Can't we just say that we have a mapping `VKeyG \mapsto Ppm` and then simply compare (and count) the `Ppm` maps?

Reply:
yea, that's fine too. both are equal easy to write and implement, one just gives you more options. I'm happy either way.

Reply:
Same here, but I think that this flexibility makes computation of agreement a bit more complex.